frontend/src/advising/ManageStudents/ExpandedStudentView.tsx

```diff
@@ -159,3 +241,105 @@ export const ExpandedStudentView: React.FC = () => {
     </Container>
   );
 };
+
+const Comments: React.FC = () => {
+  const [isExpanded, setIsExpanded] = useState(false);
+  const comments = useSelector((state: AppState) =>
+    getAdvisorCommentsFromState(state)
+  );
+
+  return (
+    <CommentsContainer>
+      <CommentHeaderWithDropDown>
+        <div
+          onClick={() => {
+            setIsExpanded(!isExpanded);
+          }}
+          style={{ marginRight: 4, marginLeft: 2 }}
+        >
+          {isExpanded ? <KeyboardArrowUpIcon /> : <KeyboardArrowDownIcon />}
+        </div>
+        <CommentsHeader>
+          <CommentHeaderText> Comments </CommentHeaderText>
+        </CommentsHeader>
+      </CommentHeaderWithDropDown>
+      {isExpanded && (
+        <div>
+          {comments.map((comment: IComment) => (
+            <Comment
+              author={comment.author}
+              comment={comment.comment}
+              createdAt={comment.createdAt}
+              updatedAt={comment.updatedAt}
+            ></Comment>
+          ))}
+          <CommentInput />
+        </div>
+      )}
+    </CommentsContainer>
+  );
+};
+
+const Comment: React.FC<IComment> = (props: IComment) => {
+  const { author, comment, createdAt, updatedAt } = props;
+  return (
+    <CommentHolderBody>
+      <CommentHeader>
+        <CommentAuthor>{author}</CommentAuthor>
+        <CommentTimestamp>{timeago.format(createdAt)}</CommentTimestamp>
+      </CommentHeader>
+      {comment}
+    </CommentHolderBody>
+  );
+};
+
+const CommentInput: React.FC = () => {
+  const [comment, setComment] = useState("");
+  const buttonDisabled = !comment;
+
+  const CommentButton = GenericColorButton(
+    "rgba(21, 116, 62, 0.68)",
+    "rgba(21, 116, 62, 0.74)"
+  );
+
+  const { planId, userId, userName } = useSelector((state: AppState) => ({
+    planId: safelyGetActivePlanIdFromState(state),
+    userId: safelyGetUserIdFromState(state),
+    userName: getUserFullNameFromState(state),
+  }));
+
+  const dispatch = useDispatch();
+
+  const handleCommentButtonClick = () => {
+    if (planId && userId) {
+      sendComment(
+        planId,
+        userId,
+        userName,
+        comment
+      ).then((response: IComment) => dispatch(addCommentAction(response)));
```

Comment:
When I send a comment from an advisor, it shows up as from the student.

Reply:
I think userId and userName have to be derived from the advisor state. The "user" in redux right now is being set to the student you are viewing.